### DISCUSSION TITLE
Gap in the bucket

### DISCUSSION THREAD

**Discussion by Mojito2110** [thanos-io/thanos#4725] [Questions & Answers]
Hi Thanos users 

I made a POC with Thanos and let it run by itself for a while without worrying about it :-)
Yesterday when I went to look in the bucket viewer web UI, I noticed gaps over a 48 hour period in the bucket every 12 days?

Can someone explain why and how to fix this?

I am using Thanos version 0.19. 
![image](https://user-images.githubusercontent.com/15526401/135646884-7435407f-48aa-432e-a9c9-18a566b02a8f.png)

**Comment by yeya24** [comment]
Hi, can you please add your Thanos configuration? It is hard to help you debug without more information.

**Comment by Mojito2110** [comment]
Hi yeya24

sure here the configurations : 

Compactor configuration  : 
./thanos compact --data-dir /srv/prometheus/thanos/compactor/data \
                 --http-address 0.0.0.0:19191 \
                 --objstore.config-file storage.yml \
                 --retention.resolution-raw 7d \
                 --retention.resolution-5m 15d \
                 --retention.resolution-1h 180d 

The compactor is lauched evey 6 hours with a cronjob 


Sidecar Configuration : 
./thanos sidecar --tsdb.path=/srv/prometheus/tsdb --prometheus.url=http://localhost:9090 --http-address=0.0.0.0:19191 --grpc-address=0.0.0.0:19090 --objstore.config-file=storage.yml



In the web bucket UI, there are 2 lines because there are 2 sources : 
The first blocks line are data block uploaded by a Sidecar
The second line are data block send by a Receive. The component not push since July

**Comment by Mojito2110** [comment]
Someone have an idea about this periodic gap ??  Do you want more information ?
